### PR TITLE
jewel: cephfs: client::mkdirs not handle well when two clients send mkdir request for a same dir

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/20823

### CODE REVIEW

**Comment by smithfarm** [comment]
retest this please

**Comment by smithfarm** [comment]
jenkins re-test this please

**Comment by smithfarm** [comment]
retest this please

**Comment by smithfarm** [comment]
```
client/Client.cc: In member function ‘int Client::mkdirs(const char*, mode_t)’:
client/Client.cc:6313:64: error: ‘perms’ was not declared in this scope
       r = _lookup(cur.get(), path[i].c_str(), CEPH_CAP_AUTH_SHARED, &next, perms);
                                                                ^
Makefile:23016: recipe for target 'client/Client.lo' failed
```

**Comment by smithfarm** [comment]
@batrick Could you take a look at the build failure? Is it trivial to fix?

File: src/client/Client.cc [hunk 1]
```diff
@@ -6302,8 +6302,13 @@ int Client::mkdirs(const char *relpath, mode_t mode)
     }
     //make new dir
     r = _mkdir(cur.get(), path[i].c_str(), mode, uid, gid, &next);
+
     //check proper creation/existence
-    if (r < 0) return r;
+    if(-EEXIST == r && i < path.depth() - 1) {
+      r = _lookup(cur.get(), path[i].c_str(), CEPH_CAP_AUTH_SHARED, &next, perms);
```
Comment (batrick) [inline src/client/Client.cc]: Yes, the conflict is trivial. Just replaces `perms` with `uid, gid`.

**Comment by smithfarm** [comment]
@batrick Thanks, man! Please have another look.

**Comment by smithfarm** [comment]
This passed an fs suite at http://tracker.ceph.com/issues/21742#note-18

**Pull request merged**